webrender/res/ps_radial_gradient.fs.glsl

```diff
@@ -55,13 +44,14 @@ void main(void) {
         }
     }
 
-    oFragColor = mix(vColors[0],
-                     vColors[1],
-                     linearStep(offset(0), offset(1), x));
+    // Either saturate or modulo the offset depending on repeat mode, then scale to number of
+    // gradient color entries.
+    x = mix(clamp(x, 0.0, 1.0), fract(x), vGradientRepeat) * vGradientScale;
 
-    for (int i=1 ; i < vStopCount-1 ; ++i) {
-        oFragColor = mix(oFragColor,
-                         vColors[i+1],
-                         linearStep(offset(i), offset(i+1), x));
-    }
+    // Start at the center of first color in the nearest 2-color entry, then offset with the
+    // fractional remainder to interpolate between the colors.
+    x = 2.0 * floor(x) + 0.5 + fract(x);
+
+    // Normalize the texture coordates so we can use texture() for bilinear filtering.
+    oFragColor = texture(sGradients, vec2(x, vGradientIndex) / vec2(textureSize(sGradients, 0)));
```

Comment:
I prefer to keep this the lowest spec feature that will implement it, as it is the most likely to perform well and be bug-free.